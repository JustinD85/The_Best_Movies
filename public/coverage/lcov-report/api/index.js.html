<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for api/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">api</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>28/28</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>8/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>23/23</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import apiKey from './movie_api_key'
&nbsp;
// const dbUrl = 'http://localhost:3000/api/'
const dbUrl = 'https://falsemotive.io/api/'
const moviesUrl = 'https://api.themoviedb.org/3/movie/'
export const options = (method, data) =&gt; ({
  method,
  headers: {
    'Content-Type': 'application/json',
  },
  body: JSON.stringify(data)
})
&nbsp;
export const postData = async (path, data) =&gt; {
  const response = await fetch(`${dbUrl}${path}`, options('POST', data))
  if (!response.ok) throw new Error(`Could not post`)
  const unfilteredData = await response.json()
  return unfilteredData.data
}
&nbsp;
export const deleteData = async (path, data) =&gt; {
  const response = await fetch(`${dbUrl}${path}`, options('DELETE', data))
  if (!response.ok) throw new Error(`Could not delete`)
  const unfilteredData = await response.json()
  return unfilteredData.results
}
&nbsp;
export const getData = async (path) =&gt; {
  const response = await fetch(`${dbUrl}${path}`)
  if (!response.ok) throw new Error(`Could not get Data`)
  const unfilteredData = await response.json()
  return unfilteredData.data
}
&nbsp;
export const getMovies = async (path) =&gt; {
  const response = await fetch(`${moviesUrl}${path}${apiKey}`)
  if (!response.ok) throw new Error('Could not get Movie')
  const unfilteredMovies = await response.json()
  return unfilteredMovies.results
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 06 2019 09:52:14 GMT-0700 (Mountain Standard Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
