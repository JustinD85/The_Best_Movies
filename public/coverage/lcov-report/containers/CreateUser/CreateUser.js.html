<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for containers/CreateUser/CreateUser.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">containers/CreateUser</a> CreateUser.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">81.25% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>13/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">81.25% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>13/16</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react'
import { Redirect, Link } from 'react-router-dom'
import { postData } from '../../api'
import PropTypes from 'prop-types'
import './CreateUser.scss'
&nbsp;
class CreateUser extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      email: '',
      password: '',
      didPost: false,
      error: ''
    }
  }
&nbsp;
  handleChange = <span class="fstat-no" title="function not covered" >(e</span>) =&gt; {
    const { name, value } = <span class="cstat-no" title="statement not covered" >e.target;</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    this.setState({</span>
      [name]: value,
    })
  }
&nbsp;
  handleSubmit = async (e) =&gt; {
    e.preventDefault();
    const { name, email, password } = this.state;
    const user = { name, email, password };
&nbsp;
    try {
      await postData('users/new', user)
      this.setState({
        didPost: true
      })
    } catch (error) {
<span class="cstat-no" title="statement not covered" >      this.setState({</span>
        error: 'User already exists'
      })
    }
  }
  componentDidMount = () =&gt; this.props.setPopup(true)
  render() {
    const { didPost, error } = this.state
    return (
      didPost ?
        &lt;Redirect to='/login' /&gt; :
        &lt;div className='CreateUser'&gt;
          &lt;form onSubmit={this.handleSubmit}&gt;
            &lt;h1 className='create-your-account'&gt;Create Your Account&lt;/h1&gt;
            &lt;input
              className='user-input'
              placeholder='Name'
              required type='text'
              onChange={this.handleChange}
              name='name' /&gt;
            &lt;input
              className='user-input'
              placeholder='Email'
              required type='email'
              onChange={this.handleChange}
              name='email' /&gt;
            &lt;input
              className='user-input'
              placeholder='Password'
              required type='password'
              onChange={this.handleChange}
              name='password' /&gt;
            &lt;button
              className='submit-user'
              type='submit'
            &gt;Submit
              &lt;/button&gt;
            &lt;Link to='/' onClick={() =&gt; this.props.setPopup(false)}&gt;Back&lt;/Link&gt;
            {error &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >&lt;h3&gt;{error}&lt;/h3&gt;}</span>
          &lt;/form&gt;
&nbsp;
        &lt;/div&gt;
    )
  }
}
&nbsp;
export default CreateUser;
&nbsp;
CreateUser.propTypes = {
  setPopup: PropTypes.func,
}</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 06 2019 09:52:14 GMT-0700 (Mountain Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
