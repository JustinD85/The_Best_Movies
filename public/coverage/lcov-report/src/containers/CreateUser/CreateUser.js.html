<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/containers/CreateUser/CreateUser.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/containers/CreateUser</a> CreateUser.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/13</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react'
import { Redirect } from 'react-router-dom'
import { postData } from '../../api/api'
&nbsp;
class CreateUser extends Component {
<span class="fstat-no" title="function not covered" >  co</span>nstructor() {
<span class="cstat-no" title="statement not covered" >    super();</span>
<span class="cstat-no" title="statement not covered" >    this.state = {</span>
      name: '',
      email: '',
      password: '',
      didPost: false,
      error: ''
    }
  }
&nbsp;
  handleChange = <span class="fstat-no" title="function not covered" >(e</span>) =&gt; {
    const { name, value } = <span class="cstat-no" title="statement not covered" >e.target;</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    this.setState({</span>
      [name]: value,
    })
  }
&nbsp;
  handleSubmit = <span class="fstat-no" title="function not covered" >as</span>ync (e) =&gt; {
<span class="cstat-no" title="statement not covered" >    e.preventDefault();</span>
    const { name, email, password } = <span class="cstat-no" title="statement not covered" >this.state;</span>
    const user = <span class="cstat-no" title="statement not covered" >{ name, email, password };</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    try {</span>
<span class="cstat-no" title="statement not covered" >      await postData('users/new', user)</span>
<span class="cstat-no" title="statement not covered" >      this.setState({</span>
        didPost: true
      })
    } catch (error) {
<span class="cstat-no" title="statement not covered" >      this.setState({</span>
        error: error.message
      })
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  re</span>nder() {
    const { didPost, error } = <span class="cstat-no" title="statement not covered" >this.state</span>
<span class="cstat-no" title="statement not covered" >    return (</span>
      didPost ?
        &lt;Redirect to='/' /&gt;
        :
        &lt;form&gt;
          &lt;input required onChange={this.handleChange} name='name' /&gt;
          &lt;input required onChange={this.handleChange} name='password' /&gt;
          &lt;input required onChange={this.handleChange} name='email' /&gt;
          &lt;button type='submit' onClick={this.handleSubmit}&gt;Submit&lt;/button&gt;
          {error &amp;&amp; &lt;h3&gt;{error}&lt;/h3&gt;}
        &lt;/form&gt;
&nbsp;
    )
  }
}
&nbsp;
export default CreateUser;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jan 30 2019 20:31:54 GMT-0700 (Mountain Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
