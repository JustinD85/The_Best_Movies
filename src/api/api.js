// const url = 'http://localhost:3000/api/'
const url = 'https://falsemotive.io/api/'

export const postData = async (options = '', user) => {
  const response = await fetch(`${url}${options}`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(user)
  })

  if (response.ok) {
    return response;
  } else {
    throw new Error(`Error posting data: ${response.status}`);
  }
}

export const fetchData = async (options) => {
  const response = await fetch(`${url}${options}`)
  if (response.ok) {
    return await response.json()
  } else {
    throw new Error('Error fetching data')
  }
}