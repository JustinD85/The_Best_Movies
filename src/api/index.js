import apiKey from './movie_api_key'

// const dbUrl = 'http://localhost:3000/api/'
const dbUrl = 'https://falsemotive.io/api/'
const moviesUrl = 'https://api.themoviedb.org/3/movie/'
export const options = (method, data) => ({
  method,
  headers: {
    'Content-Type': 'application/json',
  },
  body: JSON.stringify(data)
})

export const postData = async (path = '', data) => {
  const response = await fetch(`${dbUrl}${path}`, options('POST', data))
  if (!response.ok) throw new Error(`Could not post`)
  const unfilteredData = await response.json()
  return unfilteredData.data
}

export const deleteData = async (path = '', data) => {
  const response = await fetch(`${dbUrl}${path}`, options('DELETE', data))
  if (!response.ok) throw new Error(`Could not delete`)
  const unfilteredData = await response.json()
  return unfilteredData.results
}

export const getData = async (path) => {
  const response = await fetch(`${dbUrl}${path}`)
  if (!response.ok) throw new Error(`${response.statusText}`)
  const unfilteredData = await response.json()
  return unfilteredData.data
}

export const getMovies = async (path) => {
  const response = await fetch(`${moviesUrl}${path}${apiKey}`)
  if (!response.ok) throw new Error(`${response.statusText}`)
  const unfilteredMovies = await response.json()
  return unfilteredMovies.results
}
