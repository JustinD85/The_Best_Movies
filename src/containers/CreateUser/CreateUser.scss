.CreateUser {
  display: flex;
  height: 50%;
  width: 72%;
  z-index: 1;
  position: absolute;
  left: 22%;
  top: 10%;
  border-radius: 15px;
  background: rgba(0, 0, 0, 0.9);
  color: #ECEBEE;
  padding: 10px 30px 20px 40px;

  .create-your-account {
    color: #0CD0FC;
  }
  
  form {
    margin: auto;
    display: flex;
    flex-direction: column;
    width: 80%;
  }

  .user-input {
    font-size: 14px;
    border-radius: 5px;
    padding: 6px;
    margin: 10px;
  }

  .submit-user {
    width: 30%;
    padding: 5px;
    border-radius: 10px;
    margin: 10px auto;
  }

  a {
    color: #0CD0FC;
  }
}