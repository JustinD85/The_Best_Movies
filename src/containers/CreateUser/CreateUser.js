import React, { Component } from 'react'
import { Redirect, Link } from 'react-router-dom'
import { postData } from '../../api'
import PropTypes from 'prop-types'
import './CreateUser.scss'

class CreateUser extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      email: '',
      password: '',
      didPost: false,
      error: ''
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target;

    this.setState({
      [name]: value,
    })
  }

  handleSubmit = async (e) => {
    e.preventDefault();
    const { name, email, password } = this.state;
    const user = { name, email, password };

    try {
      await postData('users/new', user)
      this.setState({
        didPost: true
      })
    } catch (error) {
      this.setState({
        error: 'User already exists'
      })
    }
  }
  componentDidMount = () => this.props.setPopup(true)
  render() {
    const { didPost, error } = this.state
    return (
      didPost ?
        <Redirect to='/login' /> :
        <div className='CreateUser'>
          <form onSubmit={this.handleSubmit}>
            <h1 className='create-your-account'>Create Your Account</h1>
            <input
              className='user-input'
              placeholder='Name'
              required type='text'
              onChange={this.handleChange}
              name='name' />
            <input
              className='user-input'
              placeholder='Email'
              required type='email'
              onChange={this.handleChange}
              name='email' />
            <input
              className='user-input'
              placeholder='Password'
              required type='password'
              onChange={this.handleChange}
              name='password' />
            <button
              className='submit-user'
              type='submit'
            >Submit
              </button>
            <Link to='/' onClick={() => this.props.setPopup(false)}>Back</Link>
            {error && <h3>{error}</h3>}
          </form>

        </div>
    )
  }
}

export default CreateUser;

CreateUser.propTypes = {
  setPopup: PropTypes.func,
}