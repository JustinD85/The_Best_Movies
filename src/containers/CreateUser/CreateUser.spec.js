import React from 'react'
import { shallow } from 'enzyme'
import CreateUser from './CreateUser'
import { postData } from '../../api'
jest.mock('../../api')

let wrapper
let setPopup = jest.fn()
it('should match the snapshot', () => {
  wrapper = shallow(<CreateUser setPopup={jest.fn()} />)
  expect(wrapper).toMatchSnapshot()
})

it('should call setPopup if the link is clicked', () => {
  wrapper = shallow(<CreateUser setPopup={setPopup} />)
  wrapper.find('Link').simulate('click')
  expect(setPopup).toHaveBeenCalled()
})


it('should set didPost to true when a submission is sent to form', async() => {
  wrapper = shallow(<CreateUser setPopup={jest.fn()} />)
  wrapper.instance().handleSubmit({ preventDefault: () => '' })
  await expect(postData).toHaveBeenCalledTimes(1)
  await expect(wrapper.state('didPost')).toEqual(true)
})
