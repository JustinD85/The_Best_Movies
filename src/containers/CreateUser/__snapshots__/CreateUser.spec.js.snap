// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="CreateUser"
>
  <form
    onSubmit={[Function]}
  >
    <h1
      className="create-your-account"
    >
      Create Your Account
    </h1>
    <input
      className="user-input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      required={true}
      type="text"
    />
    <input
      className="user-input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
    />
    <input
      className="user-input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
    />
    <button
      className="submit-user"
      type="submit"
    >
      Submit
    </button>
    <Link
      onClick={[Function]}
      replace={false}
      to="/"
    >
      Back
    </Link>
  </form>
</div>
`;
